contrib/Trivy.gitlab-ci.yml

```diff
@@ -0,0 +1,28 @@
+Trivy_container_scanning:
+  stage: test
+  image:
+    name: alpine
+  variables:
+    # Override the GIT_STRATEGY variable in your `.gitlab-ci.yml` file and set it to `fetch` if you want to provide a `clair-whitelist.yml`
+    # file. See https://docs.gitlab.com/ee/user/application_security/container_scanning/index.html#overriding-the-container-scanning-template
+    # for details
+    GIT_STRATEGY: none
+    IMAGE: "$CI_REGISTRY_IMAGE:$CI_COMMIT_SHA"
+  allow_failure: true
+  before_script:
+    - apk add --no-cache curl docker-cli
+    - docker login -u "$CI_REGISTRY_USER" -p "$CI_REGISTRY_PASSWORD" $CI_REGISTRY
+    - curl -sfL https://raw.githubusercontent.com/aquasecurity/trivy/master/contrib/install.sh | sh -s -- -b /usr/local/bin
+    - curl -sSL -o /tmp/trivy-gitlab.tpl https://github.com/aquasecurity/trivy/raw/master/contrib/gitlab.tpl
```

Comment:
is there a way we could not fetch this from master? If the template changes and the downloaded binary does not support it, this could cause incompatibilities.

Maybe we can parse $(trivy --version) and pull from the tag instead?

Reply:
You are right. We should use environment variables to fix. With environment variables with the default value of Trivy version, (GitLab CI) users can inject arbitrary version of (Trivy-GitLab) template, which will reduce regression and be helpful for users of non-latest version GitLab.